### ISSUE TITLE
Suggestions for Search Results Filtering (Checkboxes Are Invisible And Imply Multiple Filters, and Filters Are Applied Instantly)

### ISSUE THREAD

**Issue by Santoz2** [ManifoldScholar/manifold#2759]
Manifold's Reader Interface Search Results have a section for "Search Within:" and "Show Results For:". This is reminiscent of search filters. There are a few implementation ideas that would improve both the usability and accessibility of this feature.

Notably, the implementation of switching filters is a bit odd. Visually, there is little indication that a user can interact with these filters. Without a box to contain checkboxes, it appears like these filters are static text, and not interactable elements. An empty box does appear when a keyboard user focuses these filters, but might not realize they could focus the element in the first place, and a visual mouse user never sees this, so they might not realize they can interact with this at all.

Further, while this is implemented as a group of checkboxes, it does not truly behave as such. Checkboxes typically allow users to select multiple options at once, but this instance only allows one item to be selected. As such, this may be better implemented as a group of radio-buttons, which imply that a user can only select one option. Standard checkbox implementation may be found here: [https://www.w3.org/TR/wai-aria-practices/examples/checkbox/checkbox-1/checkbox-1.html](https://www.w3.org/TR/wai-aria-practices/examples/checkbox/checkbox-1/checkbox-1.html)

Finally, this implementation of filters applies the filters as soon as a filter is selected. As such, a user may select a filter without meaning to do so, and lose track of their search results. It would be better for a user to be allowed to choose their filters, and apply those filters via an "Apply Filters" button, so that their filters are applied when they mean to apply them. An example for why this would make sense can be seen through the fact that there are two different filters. A user may want to change both filters, but the first filter they change would be applied as soon as they change it, so the search would have to run twice before the user reaches their desired results.

## Expected Behavior
Checkboxes should allow users to select multiple items. If a user can only select one item, a group of radio buttons should be used instead. The boxes that contains checks should be visible to imply to users that they can interact with this widget. A button should be provided to allow users to apply their filters, and filters should not be applied on changing a filter.

## Current Behavior
Filters are implemented as checkboxes, but the boxes are invisible so a user does not know they can interact with them. Further, a user can only select one option from these checkboxes, though checkboxes imply a user can select multiple items. Finally, filters are applied as soon as a user selects a different filter, which could cause multiple unintended search submissions that could confuse or slow down users.

The following image shows a visible box when a user places keyboard focus on the box. Without that box, it appears like the user can't interact with this widget:

![image](https://user-images.githubusercontent.com/43553368/83064790-3ad5bf00-a017-11ea-8f30-db206f849e65.png)

## Steps to Reproduce
1. Navigate to the Reader Interface
2. Navigate to a project with readable and searchable text
3. Navigate into a text contained in that project
4. Navigate to and activate the search button in the menu
5. Enter any search criteria into the edit field and submit the search
6. Navigate to the "Search Within" and "Show Results For" sections. Notice that these are implemented as checkboxes, but only one item can be selected, it does not visually appear interactable, and filters are applied immediately.

## Testing Environment
NVDA 2020.1
ESR Firefox 68.8.0 esr

**Comment by dananjohnson** [comment]
Good catch, @Santoz2! I'm seeing two issues here: first, those inputs need to be radios rather than checkboxes, as you suggested; second, it looks like there's a design regression on the search results page that makes those fields invisible unless checked. Will fix!

**Comment by dananjohnson** [comment]
@Santoz2 after closer investigation, I found it necessary to use two different kinds of controls to replace these checkboxes. You can see my solution at #2906.
